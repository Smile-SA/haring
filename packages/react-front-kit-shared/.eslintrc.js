module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
