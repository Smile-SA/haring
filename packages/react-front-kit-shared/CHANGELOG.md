# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
