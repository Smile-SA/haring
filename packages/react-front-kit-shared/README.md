# React Front Kit shared

Shared components and functions.

## Documentation

You can access the documentation here: https://smile-sa.github.io/react-front-kit-doc

## Prerequisite

You must have React 18 or later installed on your project.

## Installation

Install `@smile/react-front-kit-shared` and peer dependencies with:

```bash
npm i @smile/react-front-kit-shared @mantine/core @mantine/hooks @phosphor-icons/react
```

## Styles

Import the styles in your app with:

```js
import '@mantine/core/styles.css';
import '@smile/react-front-kit-shared/style.css';
```

## License

LGPL-3.0
