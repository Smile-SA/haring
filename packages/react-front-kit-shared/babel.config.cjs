const { babelConfig } = require('test');

module.exports = babelConfig;
