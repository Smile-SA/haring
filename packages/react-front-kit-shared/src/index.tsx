/* eslint-disable react-refresh/only-export-components */
'use client';

// component exports
export { ColumnPlus, FolderMove } from './1-styleGuide/Icons';
export { FileIcon } from './1-styleGuide/FileIcon/FileIcon';
export type { IFileIconProps } from './1-styleGuide/FileIcon/FileIcon';
export { Provider } from './3-custom/Provider/Provider';
export type { IProviderProps } from './3-custom/Provider/Provider';
// context exports
export {
  themeContext,
  useMainTheme,
  usePrimaryTheme,
  useSecondaryTheme,
  useThemes,
} from './contexts';
// helper exports
export {
  createThemes,
  isCallback,
  isNotNullNorEmpty,
  typeGuard,
  typeGuardInterface,
} from './helpers';
export type {
  IGuardedType,
  IPrimitiveOrConstructor,
  ITypeGenericInterface,
  ITypeMap,
} from './helpers';
// type exports
export type {
  IAction,
  IActionConfirmModalProps,
  IConfirmAction,
  IConfirmModal,
  IOption,
  IOptions,
  IThemeOverride,
  IThemes,
} from './types';
// type exports
export { mainTheme, primaryTheme, secondaryTheme, themes } from './theme';
