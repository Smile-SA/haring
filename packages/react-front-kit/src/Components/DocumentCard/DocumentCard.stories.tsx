import type { Meta, StoryObj } from '@storybook/react';

import { DocumentCard as Cmp } from './DocumentCard';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/DocumentCard',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const DocumentCard: IStory = {
  args: {
    author: 'Aline',
    children: 'Children',
    date: 'Published on December 24, 2023',
    iconType: 'PDF',
    path: '(Customer > 567890456 > Invoices)',
    title: 'Random_File.PDF',
  },
};
