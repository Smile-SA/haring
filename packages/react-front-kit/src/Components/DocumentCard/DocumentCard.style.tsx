import { createStyles } from '@mantine/styles';

export const useStyles = createStyles((theme) => ({
  author: {
    fontWeight: 600,
  },
  children: {
    marginTop: '16px',
  },
  header: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  headerBottom: {
    '@media (max-width: 833px)': {
      flexDirection: 'column',
    },
    color: theme.colors.dark[3],
    display: 'flex',
    fontSize: '14px',
    marginRight: '12px',
  },
  headerTop: {
    '@media (max-width: 833px)': {
      flexDirection: 'column',
    },
    display: 'flex',
    flexWrap: 'wrap',
  },
  iconContainer: {
    marginRight: '16px',
  },
  image: {
    border: `1px solid ${theme.colors.gray[2]}`,
    borderRadius: '16px',
    marginRight: '32px',
  },
  imageRoot: {
    '@media (max-width: 414px)': {
      maxWidth: '100%',
    },
    '@media (max-width: 833px)': {
      flexDirection: 'column',
      marginBottom: '20px',
    },
    marginRight: '32px',
    maxWidth: '165px',
  },
  path: {
    color: theme.colors.dark[6],
  },
  root: {
    cursor: 'pointer',
    display: 'flex',
    // eslint-disable-next-line sort-keys
    '@media (max-width: 833px)': {
      flexDirection: 'column',
    },
  },
  separator: {
    '@media (max-width: 833px)': {
      display: 'none',
    },
    margin: '0 12px',
  },
  title: {
    display: 'flex',
    fontSize: '18px',
    fontWeight: 'bold',
    margin: '0 16px 0 0',
  },
}));
