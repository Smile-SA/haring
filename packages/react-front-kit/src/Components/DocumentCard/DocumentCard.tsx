'use client';

import type { ReactNode } from 'react';

import { Box, Image, useMantineTheme } from '@mantine/core';
import { FileIcon } from '@smile/react-front-kit-shared';

import defaultImage from '../../../assets/defaultImage.jpg';

import { useStyles } from './DocumentCard.style';

export interface IDocumentCardProps {
  author: ReactNode;
  children?: ReactNode;
  date?: ReactNode;
  iconType?: string;
  image?: string;
  path?: ReactNode;
  title?: ReactNode;
}

export function DocumentCard(props: IDocumentCardProps): ReactNode {
  const {
    author,
    children,
    date,
    iconType,
    image = defaultImage,
    path,
    title,
  } = props;
  const { classes } = useStyles();

  const theme = useMantineTheme();

  return (
    <Box className={classes.root}>
      <Image
        classNames={{ image: classes.image, root: classes.imageRoot }}
        radius={16}
        src={image}
      />
      <div>
        <div>
          <div className={classes.header}>
            <div className={classes.iconContainer}>
              <FileIcon
                color={theme.colors.cyan[9]}
                size={38}
                type={iconType}
                weight="light"
              />
            </div>
            <div>
              <div className={classes.headerTop}>
                {Boolean(title) && (
                  <span className={classes.title}>{title}</span>
                )}
                {Boolean(path) && <span className={classes.path}>{path}</span>}
              </div>
              <div className={classes.headerBottom}>
                {Boolean(date) && (
                  <>
                    <span>{date}</span>
                    {Boolean(author) && (
                      <span className={classes.separator}>-</span>
                    )}
                  </>
                )}
                {Boolean(author) && (
                  <span className={classes.author}>{author}</span>
                )}
              </div>
            </div>
          </div>
        </div>
        <div className={classes.children}>{children}</div>
      </div>
    </Box>
  );
}
