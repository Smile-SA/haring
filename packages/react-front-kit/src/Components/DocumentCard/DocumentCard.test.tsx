import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { DocumentCard } from './DocumentCard';

describe('DocumentCard', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <DocumentCard
        author="Aline"
        date="Published on December 24, 2023"
        iconType="PDF"
        path="(Customer > 567890456 > Invoices)"
        title="Random_File.PDF"
      >
        children
      </DocumentCard>,
    );
    expect(container).toMatchSnapshot();
  });
});
