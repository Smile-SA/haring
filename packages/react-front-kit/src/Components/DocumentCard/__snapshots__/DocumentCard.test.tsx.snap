// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentCard matches snapshot 1`] = `
<div>
  <div
    class="mantine-1feahhc"
  >
    <div
      class="mantine-Image-root mantine-1hgbf0p"
      style="width: 100%;"
    >
      <figure
        class="mantine-qenwvq mantine-Image-figure"
      >
        <div
          class="mantine-1iugybl mantine-Image-imageWrapper"
        >
          <img
            class="mantine-Image-image mantine-dc8ibh"
            src=""
            style="object-fit: cover; width: 100%; height: auto;"
          />
        </div>
      </figure>
    </div>
    <div>
      <div>
        <div
          class="mantine-mdzvt7"
        >
          <div
            class="mantine-qu0qb1"
          >
            <svg
              fill="#0b7285"
              height="38"
              viewBox="0 0 256 256"
              width="38"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M222,152a6,6,0,0,1-6,6H190v20h18a6,6,0,0,1,0,12H190v18a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h32A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm84,8a34,34,0,0,1-34,34H112a6,6,0,0,1-6-6V152a6,6,0,0,1,6-6h16A34,34,0,0,1,162,180Zm-12,0a22,22,0,0,0-22-22H118v44h10A22,22,0,0,0,150,180ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
              />
            </svg>
          </div>
          <div>
            <div
              class="mantine-dxs7ne"
            >
              <span
                class="mantine-fakv65"
              >
                Random_File.PDF
              </span>
              <span
                class="mantine-xrrpki"
              >
                (Customer &gt; 567890456 &gt; Invoices)
              </span>
            </div>
            <div
              class="mantine-1fna5yi"
            >
              <span>
                Published on December 24, 2023
              </span>
              <span
                class="mantine-mj4lz4"
              >
                -
              </span>
              <span
                class="mantine-l9up18"
              >
                Aline
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mantine-j5huwo"
      >
        children
      </div>
    </div>
  </div>
</div>
`;
